### ISSUE TITLE
Backward compatibility tests do not run with the security plugin enabled

### ISSUE THREAD

**Issue by peternied** [opensearch-project/security#3056]
### Issue
When BWC tests are executing they call `GET _node/{NODE_ID}/plugins` and check to see if the security plugin is in the list.  This doesn't check that the securty plugin is operational.  After inspection, it is not operational.  The BWC tests should run with all features of the security plugin enabled otherwise we cannot determine if backwards incompatible changes are being assessed.

 - Related to #3053

### Sub issues to address
- [x] https://github.com/opensearch-project/OpenSearch/pull/8900
- [x] Enable security plugin on BWC tests
- [ ] Correct audit logging source for tests
- [ ] Modify the both the client and admin client to use SSL connection
- [x] Update the test case to allow specifying the use that is used to make the connection
- [ ] (Recommended) Add a test case that calls 'whoami' and confirms the identity of the caller

### Additional Context

I'm seeing the following line that seems troublesome, this would imply that the security plugin features won't be available, doing more digging

>» WARN ][o.o.s.OpenSearchSecurityPlugin] [securityBwcCluster1-0] OpenSearch Security plugin installed but disabled. This can expose your configuration (including passwords) to the public.

After looking at the configuration settings, (Thanks for paired debugging with me @parasjain1) these test don't actually start up the security plugin at all, the following configuration line needs to be switched from false -> true.

https://github.com/opensearch-project/security/blob/8063e1b64f880afb7dc54311f3110b162b913038/bwc-test/build.gradle#L128

However, after attempting to startup the node, copying configuration from [SecureRestClientBuilder.java]( https://github.com/opensearch-project/common-utils/blob/6b783645f23e9b9a40cd0ceb0a3859c6763e3bd7/src/main/java/org/opensearch/commons/rest/SecureRestClientBuilder.java), there are still many errors coming in from the cluster:

> » ERROR][o.o.s.s.h.n.SecuritySSLNettyHttpServerTransport] [securityBwcCluster1-0] Exception during establishing a SSL connection: io.netty.handler.ssl.NotSslRecordException: not an SSL/TLS record: 4...0a
»  io.netty.handler.ssl.NotSslRecordException: not an SSL/TLS record: 4..a
»  	at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1225) ~[netty-handler-4.1.94.Final.jar:4.1.94.Final]

These errors are happening because the cluster startup check is hardcoded to http, need to make this parameterized

> WaitForHttpResource wait = new WaitForHttpResource("http", getFirstNode().getHttpSocketURI(), nodes.size());
[[Source]](https://github.com/opensearch-project/OpenSearch/blob/99f28cb6f6bd73a6d832c6bbb88e5b9c2ccdb139/buildSrc/src/main/java/org/opensearch/gradle/testclusters/OpenSearchCluster.java#L557)

Branch that can be used to kickstart this effort https://github.com/opensearch-project/security/compare/main...peternied:security:bwc-ssl

**Comment by scrawfor99** [comment]
I think this is the general idea for core: https://github.com/opensearch-project/OpenSearch/pull/8900. I am not sure about actually inputting the setting however since I don't see analogs for the type of setting I am creating here. I will have to look into it more.

**Comment by scrawfor99** [comment]
[Triage] @scrawfor99 is handling this issue.

**Comment by scrawfor99** [comment]
https://github.com/opensearch-project/OpenSearch/pull/8900 addresses part 1 of this issue. 

https://github.com/scrawfor99/security/tree/testClusterChanges outlines some of the changes likely required for the security side of things.

**Comment by scrawfor99** [comment]
Not sure how to diagnose this yet: file:///Users/steecraw/security/bwc-test/build/reports/tests/securityBwcCluster%23oldVersionClusterTask1/classes/org.opensearch.security.bwc.SecurityBackwardsCompatibilityIT.html

**Comment by peternied** [comment]
@scrawfor99 Do you have a pull request/branch where you get that reproduction when attempting to start the bwc tests?  Let me know and I'll take a look

**Comment by peternied** [comment]
Here is how I set up my machine, and I can see those errors...
```
cd ~/git/OpenSearch/
git remote add scrawfor99 https://github.com/scrawfor99/OpenSearch.git
git fetch scrawfor99
git checkout scrawfor99/bwcFix
git merge origin/main
./gradlew build-tools:publishToMavenLocal
./gradlew distribution:archives:linux-tar:assemble

cd ../security
git remote add scrawfor99 https://github.com/scrawfor99/OpenSearch.git
git checkout scrawfor99/testClusterChanges
git merge origin/main
./gradlew assemble
mkdir -p ${GIT_PROJECT_ROOT}/security/bwc-test/src/test/resources/3.0.0.0
cp ${GIT_PROJECT_ROOT}/security/build/distributions/opensearch-security-3.0.0.0-SNAPSHOT.zip ${GIT_PROJECT_ROOT}/security/bwc-test/src/test/resources/3.0.0.0/opensearch-security-3.0.0.0-SNAPSHOT.zip
./gradlew -p bwc-test bwcTestSuite -Dtests.opensearch.http.protocol=https -Dtests.opensearch.username=admin -Dtests.opensearch.password=admin -PcustomDistributionUrl="${GIT_PROJECT_ROOT}/opensearch/distribution/archives/linux-tar/build/distributions/opensearch-min-3.0.0-SNAPSHOT-linux-x64.tar.gz" -i
```

**Comment by peternied** [comment]
Managed to get some of the scenarios running, still sees some errors, here is the commit that does the most to get that unblocked https://github.com/scrawfor99/security/commit/568c468a42699e84634c897d5dc5f5586edeee1e

Functional branch - basic readme on how to get started https://github.com/peternied/security/blob/testClusterChanges/bwc-test/README.md

#### Unblocked
- securityBwcCluster#fullRestartClusterTask
- securityBwcCluster#oldVersionClusterTask0
- securityBwcCluster#oldVersionClusterTask1

### Still broken
- securityBwcCluster#mixedClusterTask

```
Execution failed for task ':securityBwcCluster#mixedClusterTask'.
> `cluster{::securityBwcCluster0}` failed to wait for cluster health yellow after 40 SECONDS
    IO error while waiting cluster
    503 Service Unavailable
```

**Comment by peternied** [comment]
@scrawfor99 Since we have some of the tests working could you file a separate issue to track the fixing mixed cluster task?  Seems like the backporting the test harness fixes into 2.X might be needed, or maybe there is something else wrong.  I didn't dive in more, but cluster not starting up could be due to an issue with the nodes connecting to the cluster or something else that needs more digging.

**Comment by scrawfor99** [comment]
Hi @peternied, no problem. I am not sure what this will mean for this issue though... At what point will we be able to consider it done do you think?

**Comment by peternied** [comment]
I'm not worried about the meta issue vs actual issues so long as the we can unblock https://github.com/opensearch-project/security/pull/2802

I was thinking of having a separate issue because maybe there is some work that can happen in parallel before we've got everything in place.  If your instincts say to keep it one issue lets do it.

**Comment by scrawfor99** [comment]
Currently looking into how normal upgrades work because they must handle plugins somehow and we want to mimic that in the test handling in core.

**Comment by scrawfor99** [comment]
Failure in mixedClusterTest is actually from this:

```
[2023-08-16T16:51:10,047][INFO ][o.o.s.a.i.AuditLogImpl   ] [securityBwcCluster1-2] .opendistro_security is used as internal security index.
[2023-08-16T16:51:10,047][INFO ][o.o.s.a.i.AuditLogImpl   ] [securityBwcCluster1-2] Internal index used for posting audit logs is null
[2023-08-16T16:51:10,048][INFO ][o.o.s.c.ConfigurationRepository] [securityBwcCluster1-2] Hot-reloading of audit configuration is enabled
[2023-08-16T16:51:10,051][INFO ][o.o.s.c.ConfigurationRepository] [securityBwcCluster1-2] Node 'securityBwcCluster1-2' initialized
[2023-08-16T16:51:12,409][DEBUG][o.o.c.c.PublicationTransportHandler] [securityBwcCluster1-2] received diff cluster state version [39] with uuid [7NdxpHS5TT6jDYFoTa66_g], diff size [196]
[2023-08-16T16:51:14,680][DEBUG][o.o.c.c.LeaderChecker    ] [securityBwcCluster1-2] leader [{securityBwcCluster1-0}{UrHguxEtQG-IFR6e3IwHBw}{MXMAI8_LQ9aoCB-eBMtXRw}{127.0.0.1}{127.0.0.1:63821}{dimr}{testattr=test, upgraded=true, shard_indexing_pressure_enabled=true}] disconnected
[2023-08-16T16:51:14,682][INFO ][o.o.c.c.Coordinator      ] [securityBwcCluster1-2] cluster-manager node [{securityBwcCluster1-0}{UrHguxEtQG-IFR6e3IwHBw}{MXMAI8_LQ9aoCB-eBMtXRw}{127.0.0.1}{127.0.0.1:63821}{dimr}{testattr=test, upgraded=true, shard_indexing_pressure_enabled=true}] failed, restarting discovery
org.opensearch.transport.NodeDisconnectedException: [securityBwcCluster1-0][127.0.0.1:63821][disconnected] disconnected
[2023-08-16T16:51:14,683][DEBUG][o.o.c.c.Coordinator      ] [securityBwcCluster1-2] onLeaderFailure: coordinator becoming CANDIDATE in term 4 (was FOLLOWER, lastKnownLeader was [Optional[{securityBwcCluster1-0}{UrHguxEtQG-IFR6e3IwHBw}{MXMAI8_LQ9aoCB-eBMtXRw}{127.0.0.1}{127.0.0.1:63821}{dimr}{testattr=test, upgraded=true, shard_indexing_pressure_enabled=true}]])
[2023-08-16T16:51:14,685][INFO ][o.o.c.s.ClusterApplierService] [securityBwcCluster1-2] cluster-manager node changed {previous [{securityBwcCluster1-0}{UrHguxEtQG-IFR6e3IwHBw}{MXMAI8_LQ9aoCB-eBMtXRw}{127.0.0.1}{127.0.0.1:63821}{dimr}{testattr=test, upgraded=true, shard_indexing_pressure_enabled=true}], current []}, term: 4, version: 39, reason: becoming candidate: onLeaderFailure
[2023-08-16T16:51:14,691][WARN ][o.o.c.NodeConnectionsService] [securityBwcCluster1-2] failed to connect to {securityBwcCluster1-0}{UrHguxEtQG-IFR6e3IwHBw}{MXMAI8_LQ9aoCB-eBMtXRw}{127.0.0.1}{127.0.0.1:63821}{dimr}{testattr=test, upgraded=true, shard_indexing_pressure_enabled=true} (tried [1] times)
org.opensearch.transport.ConnectTransportException: [securityBwcCluster1-0][127.0.0.1:63821] connect_exception
	at org.opensearch.transport.TcpTransport$ChannelsConnectedListener.onFailure(TcpTransport.java:1074) ~[opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.opensearch.core.action.ActionListener.lambda$toBiConsumer$2(ActionListener.java:215) ~[opensearch-core-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.opensearch.common.concurrent.CompletableContext.lambda$addListener$0(CompletableContext.java:57) ~[opensearch-common-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841) ~[?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510) ~[?:?]
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2162) ~[?:?]
	at org.opensearch.common.concurrent.CompletableContext.completeExceptionally(CompletableContext.java:72) ~[opensearch-common-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.opensearch.transport.netty4.Netty4TcpChannel.lambda$addListener$0(Netty4TcpChannel.java:81) ~[?:?]
	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:590) ~[?:?]
	at io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:583) ~[?:?]
	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:559) ~[?:?]
	at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:492) ~[?:?]
	at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:636) ~[?:?]
	at io.netty.util.concurrent.DefaultPromise.setFailure0(DefaultPromise.java:629) ~[?:?]
	at io.netty.util.concurrent.DefaultPromise.tryFailure(DefaultPromise.java:118) ~[?:?]
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.fulfillConnectPromise(AbstractNioChannel.java:321) ~[?:?]
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:337) ~[?:?]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:776) ~[?:?]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:689) ~[?:?]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:652) ~[?:?]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562) ~[?:?]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[?:?]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[?:?]
	at java.lang.Thread.run(Thread.java:833) [?:?]
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: 127.0.0.1/127.0.0.1:63821
Caused by: java.net.ConnectException: Connection refused
	at sun.nio.ch.Net.pollConnect(Native Method) ~[?:?]
	at sun.nio.ch.Net.pollConnectNow(Net.java:672) ~[?:?]
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:946) ~[?:?]
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:337) ~[?:?]
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334) ~[?:?]
	... 7 more
[2023-08-16T16:51:14,708][ERROR][o.o.s.s.t.SecuritySSLNettyTransport] [securityBwcCluster1-2] Exception during establishing a SSL connection: java.net.SocketException: Connection reset
java.net.SocketException: Connection reset
	at sun.nio.ch.SocketChannelImpl.throwConnectionReset(SocketChannelImpl.java:394) ~[?:?]
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:426) ~[?:?]
	at org.opensearch.transport.CopyBytesSocketChannel.readFromSocketChannel(CopyBytesSocketChannel.java:155) ~[transport-netty4-client-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.opensearch.transport.CopyBytesSocketChannel.doReadBytes(CopyBytesSocketChannel.java:140) ~[transport-netty4-client-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:151) [netty-transport-4.1.96.Final.jar:4.1.96.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788) [netty-transport-4.1.96.Final.jar:4.1.96.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:689) [netty-transport-4.1.96.Final.jar:4.1.96.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:652) [netty-transport-4.1.96.Final.jar:4.1.96.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562) [netty-transport-4.1.96.Final.jar:4.1.96.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) [netty-common-4.1.96.Final.jar:4.1.96.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-common-4.1.96.Final.jar:4.1.96.Final]
	at java.lang.Thread.run(Thread.java:833) [?:?]
[2023-08-16T16:51:14,708][ERROR][o.o.s.s.t.SecuritySSLNettyTransport] [securityBwcCluster1-2] Exception during establishing a SSL connection: java.net.SocketException: Connection reset
java.net.SocketException: Connection reset
	at sun.nio.ch.SocketChannelImpl.throwConnectionReset(SocketChannelImpl.java:394) ~[?:?]
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:426) ~[?:?]
	at org.opensearch.transport.CopyBytesSocketChannel.readFromSocketChannel(CopyBytesSocketChannel.java:155) ~[transport-netty4-client-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at org.opensearch.transport.CopyBytesSocketChannel.doReadBytes(CopyBytesSocketChannel.java:140) ~[transport-netty4-client-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:151) [netty-transport-4.1.96.Final.jar:4.1.96.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788) [netty-transport-4.1.96.Final.jar:4.1.96.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:689) [netty-transport-4.1.96.Final.jar:4.1.96.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:652) [netty-transport-4.1.96.Final.jar:4.1.96.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562) [netty-transport-4.1.96.Final.jar:4.1.96.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) [netty-common-4.1.96.Final.jar:4.1.96.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-common-4.1.96.Final.jar:4.1.96.Final]
	at java.lang.Thread.run(Thread.java:833) [?:?]
``` 

The first and second nodes update fine but the final node fails to join the cluster.

**Comment by scrawfor99** [comment]
I noticed in the failing mixedClusterTest that we run into what looks like a node trying to join itself:

```

[2023-08-17T17:07:17,088][INFO ][o.o.c.c.JoinHelper       ] [securityBwcCluster0-0] failed to join {securityBwcCluster0-0}{nIIK5MlvS9KOJF81nv9v8w}{2qkpyY1iROGFeyQY4y9hIQ}{127.0.0.1}{127.0.0.1:53701}{dimr}{testattr=test, shard_indexing_pressure_enabled=true} with JoinRequest{sourceNode={securityBwcCluster0-0}{nIIK5MlvS9KOJF81nv9v8w}{2qkpyY1iROGFeyQY4y9hIQ}{127.0.0.1}{127.0.0.1:53701}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}, minimumTerm=1, optionalJoin=Optional[Join{term=2, lastAcceptedTerm=0, lastAcceptedVersion=0, sourceNode={securityBwcCluster0-0}{nIIK5MlvS9KOJF81nv9v8w}{2qkpyY1iROGFeyQY4y9hIQ}{127.0.0.1}{127.0.0.1:53701}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}, targetNode={securityBwcCluster0-0}{nIIK5MlvS9KOJF81nv9v8w}{2qkpyY1iROGFeyQY4y9hIQ}{127.0.0.1}{127.0.0.1:53701}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}}]}
org.opensearch.transport.RemoteTransportException: [securityBwcCluster0-0][127.0.0.1:53701][internal:cluster/coordination/join]
```

**Comment by scrawfor99** [comment]
First issue in updated node occurs before update: 

```
[2023-08-18T12:38:30,543][WARN ][o.o.b.Natives            ] [securityBwcCluster0-0] unable to load JNA native support library, native methods will be disabled.
java.lang.UnsatisfiedLinkError: Can't load library: /Users/steecraw/Library/Caches/JNA/temp/jna11807200050257941166.tmp
	at java.lang.ClassLoader.loadLibrary(ClassLoader.java:2393) ~[?:?]
	at java.lang.Runtime.load0(Runtime.java:755) ~[?:?]
	at java.lang.System.load(System.java:1953) ~[?:?]
	at com.sun.jna.Native.loadNativeDispatchLibraryFromClasspath(Native.java:1018) ~[jna-5.5.0.jar:5.5.0 (b0)]
	at com.sun.jna.Native.loadNativeDispatchLibrary(Native.java:988) ~[jna-5.5.0.jar:5.5.0 (b0)]
	at com.sun.jna.Native.<clinit>(Native.java:195) ~[jna-5.5.0.jar:5.5.0 (b0)]
	at java.lang.Class.forName0(Native Method) ~[?:?]
	at java.lang.Class.forName(Class.java:375) ~[?:?]
	at org.opensearch.bootstrap.Natives.<clinit>(Natives.java:60) [opensearch-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
	at org.opensearch.bootstrap.Bootstrap.initializeNatives(Bootstrap.java:123) [opensearch-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
	at org.opensearch.bootstrap.Bootstrap.setup(Bootstrap.java:191) [opensearch-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
	at org.opensearch.bootstrap.Bootstrap.init(Bootstrap.java:404) [opensearch-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
	at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:180) [opensearch-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
	at org.opensearch.bootstrap.OpenSearch.execute(OpenSearch.java:171) [opensearch-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
	at org.opensearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:104) [opensearch-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
	at org.opensearch.cli.Command.mainWithoutErrorHandling(Command.java:138) [opensearch-cli-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
	at org.opensearch.cli.Command.main(Command.java:101) [opensearch-cli-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
	at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:137) [opensearch-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
	at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:103) [opensearch-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
[2023-08-18T12:38:30,547][WARN ][o.o.b.Natives            ] [securityBwcCluster0-0] cannot check if running as root because JNA is not a
```
This seems standard however as I see it appear elsewhere on the non-upgrading nodes as well.

After that we transition into some basic config logging

```
[2023-08-18T12:38:30,968][INFO ][o.o.s.OpenSearchSecurityPlugin] [securityBwcCluster0-0] OpenSearch Config path is /Users/steecraw/security/bwc-test/build/testclusters/securityBwcCluster0-0/config
[2023-08-18T12:38:31,102][INFO ][o.o.s.s.DefaultSecurityKeyStore] [securityBwcCluster0-0] JVM supports TLSv1.3
[2023-08-18T12:38:31,103][INFO ][o.o.s.s.DefaultSecurityKeyStore] [securityBwcCluster0-0] Config directory is /Users/steecraw/security/bwc-test/build/testclusters/securityBwcCluster0-0/config/, from there the key- and truststore files are resolved relatively
[2023-08-18T12:38:31,440][INFO ][o.o.s.s.DefaultSecurityKeyStore] [securityBwcCluster0-0] TLS Transport Client Provider : JDK
[2023-08-18T12:38:31,440][INFO ][o.o.s.s.DefaultSecurityKeyStore] [securityBwcCluster0-0] TLS Transport Server Provider : JDK
[2023-08-18T12:38:31,441][INFO ][o.o.s.s.DefaultSecurityKeyStore] [securityBwcCluster0-0] TLS HTTP Provider             : JDK
[2023-08-18T12:38:31,441][INFO ][o.o.s.s.DefaultSecurityKeyStore] [securityBwcCluster0-0] Enabled TLS protocols for transport layer : [TLSv1.3, TLSv1.2]
[2023-08-18T12:38:31,441][INFO ][o.o.s.s.DefaultSecurityKeyStore] [securityBwcCluster0-0] Enabled TLS protocols for HTTP layer      : [TLSv1.3, TLSv1.2]
[2023-08-18T12:38:31,459][INFO ][o.o.s.OpenSearchSecurityPlugin] [securityBwcCluster0-0] Clustername: securityBwcCluster0
[2023-08-18T12:38:31,465][WARN ][o.o.s.OpenSearchSecurityPlugin] [securityBwcCluster0-0] Directory /Users/steecraw/security/bwc-test/build/testclusters/securityBwcCluster0-0/config has insecure file permissions (should be 0700)
[2023-08-18T12:38:31,465][WARN ][o.o.s.OpenSearchSecurityPlugin] [securityBwcCluster0-0] File /Users/steecraw/security/bwc-test/build/testclusters/securityBwcCluster0-0/config/kirk.pem has insecure file permissions (should be 0600)
[2023-08-18T12:38:31,465][WARN ][o.o.s.OpenSearchSecurityPlugin] [securityBwcCluster0-0] File /Users/steecraw/security/bwc-test/build/testclusters/securityBwcCluster0-0/config/esnode-key.pem has insecure file permissions (should be 0600)
[2023-08-18T12:38:31,465][WARN ][o.o.s.OpenSearchSecurityPlugin] [securityBwcCluster0-0] File /Users/steecraw/security/bwc-test/build/testclusters/securityBwcCluster0-0/config/root-ca.pem has insecure file permissions (should be 0600)
[2023-08-18T12:38:31,466][WARN ][o.o.s.OpenSearchSecurityPlugin] [securityBwcCluster0-0] Directory /Users/steecraw/security/bwc-test/build/testclusters/securityBwcCluster0-0/config/jvm.options.d has insecure file permissions (should be 0700)
[2023-08-18T12:38:31,466][WARN ][o.o.s.OpenSearchSecurityPlugin] [securityBwcCluster0-0] File /Users/steecraw/security/bwc-test/build/testclusters/securityBwcCluster0-0/config/kirk-key.pem has insecure file permissions (should be 0600)
[2023-08-18T12:38:31,466][WARN ][o.o.s.OpenSearchSecurityPlugin] [securityBwcCluster0-0] File /Users/steecraw/security/bwc-test/build/testclusters/securityBwcCluster0-0/config/kirk-keystore.jks has insecure file permissions (should be 0600)
[2023-08-18T12:38:31,466][WARN ][o.o.s.OpenSearchSecurityPlugin] [securityBwcCluster0-0] File /Users/steecraw/security/bwc-test/build/testclusters/securityBwcCluster0-0/config/opensearch.yml has insecure file permissions (should be 0600)
[2023-08-18T12:38:31,466][WARN ][o.o.s.OpenSearchSecurityPlugin] [securityBwcCluster0-0] File /Users/steecraw/security/bwc-test/build/testclusters/securityBwcCluster0-0/config/esnode.pem has insecure file permissions (should be 0600)
```
Again, this appears correct. 

```
[2023-08-18T12:38:31,469][INFO ][o.o.p.PluginsService     ] [securityBwcCluster0-0] loaded module [transport-netty4]
[2023-08-18T12:38:31,470][INFO ][o.o.p.PluginsService     ] [securityBwcCluster0-0] loaded plugin [opensearch-security]
[2023-08-18T12:38:31,482][INFO ][o.o.s.OpenSearchSecurityPlugin] [securityBwcCluster0-0] Disabled https compression by default to mitigate BREACH attacks. You can enable it by setting 'http.compression: true' in opensearch.yml
[2023-08-18T12:38:31,486][INFO ][o.o.e.ExtensionsManager  ] [securityBwcCluster0-0] ExtensionsManager initialized
[2023-08-18T12:38:31,505][INFO ][o.o.e.NodeEnvironment    ] [securityBwcCluster0-0] using [1] data paths, mounts [[/System/Volumes/Data (/dev/disk3s5)]], net usable_space [266.9gb], net total_space [460.4gb], types [apfs]
[2023-08-18T12:38:31,506][INFO ][o.o.e.NodeEnvironment    ] [securityBwcCluster0-0] heap size [512mb], compressed ordinary object pointers [true]
[2023-08-18T12:38:31,540][INFO ][o.o.n.Node               ] [securityBwcCluster0-0] node name [securityBwcCluster0-0], node ID [mWKwveN4RhmQ60MfGRLb6w], cluster name [securityBwcCluster0], roles [ingest, remote_cluster_client, data, cluster_manager]
[2023-08-18T12:38:32,135][WARN ][o.o.s.c.Salt             ] [securityBwcCluster0-0] If you plan to use field masking pls configure compliance salt e1ukloTsQlOgPquJ to be a random string of 16 chars length identical on all nodes
[2023-08-18T12:38:32,160][INFO ][o.o.s.a.i.AuditLogImpl   ] [securityBwcCluster0-0] Message routing enabled: true
[2023-08-18T12:38:32,179][INFO ][o.o.s.f.SecurityFilter   ] [securityBwcCluster0-0] <NONE> indices are made immutable.
[2023-08-18T12:38:32,345][INFO ][o.o.t.NettyAllocator     ] [securityBwcCluster0-0] creating NettyAllocator with the following configs: [name=unpooled, suggested_max_allocation_size=256kb, factors={opensearch.unsafe.use_unpooled_allocator=null, g1gc_enabled=true, g1gc_region_size=1mb, heap_size=512mb}]
[2023-08-18T12:38:32,406][INFO ][o.o.d.DiscoveryModule    ] [securityBwcCluster0-0] using discovery type [zen] and seed hosts providers [settings, file]
[2023-08-18T12:38:32,592][WARN ][o.o.g.DanglingIndicesState] [securityBwcCluster0-0] gateway.auto_import_dangling_indices is disabled, dangling indices will not be automatically detected or imported and must be managed manually
[2023-08-18T12:38:32,704][INFO ][o.o.n.Node               ] [securityBwcCluster0-0] initialized
[2023-08-18T12:38:32,704][INFO ][o.o.n.Node               ] [securityBwcCluster0-0] starting ...
[2023-08-18T12:38:32,767][INFO ][o.o.t.TransportService   ] [securityBwcCluster0-0] publish_address {127.0.0.1:54466}, bound_addresses {[::1]:54465}, {127.0.0.1:54466}
[2023-08-18T12:38:32,768][INFO ][o.o.t.TransportService   ] [securityBwcCluster0-0] Remote clusters initialized successfully.
[2023-08-18T12:38:32,912][WARN ][o.o.b.BootstrapChecks    ] [securityBwcCluster0-0] system call filters failed to install; check the logs and fix your configuration or disable system call filters at your own risk
[2023-08-18T12:38:32,915][DEBUG][o.o.c.c.Coordinator      ] [securityBwcCluster0-0] startInitialJoin: coordinator becoming CANDIDATE in term 0 (was null, lastKnownLeader was [Optional.empty])
[2023-08-18T12:38:32,917][WARN ][o.o.d.FileBasedSeedHostsProvider] [securityBwcCluster0-0] expected, but did not find, a dynamic hosts list at [/Users/steecraw/security/bwc-test/build/testclusters/securityBwcCluster0-0/config/unicast_hosts.txt]
[2023-08-18T12:38:32,928][INFO ][o.o.h.AbstractHttpServerTransport] [securityBwcCluster0-0] publish_address {127.0.0.1:54468}, bound_addresses {[::1]:54467}, {127.0.0.1:54468}
[2023-08-18T12:38:32,929][INFO ][o.o.n.Node               ] [securityBwcCluster0-0] started
```

Other node components are then loaded, an address is published, and the node starts. 

```
[2023-08-18T12:38:32,930][INFO ][o.o.s.c.ConfigurationRepository] [securityBwcCluster0-0] Will attempt to create index .opendistro_security and default configs if they are absent
[2023-08-18T12:38:32,931][INFO ][o.o.s.OpenSearchSecurityPlugin] [securityBwcCluster0-0] 0 OpenSearch Security modules loaded so far: []
[2023-08-18T12:38:32,931][INFO ][o.o.s.c.ConfigurationRepository] [securityBwcCluster0-0] Background init thread started. Install default config?: true
[2023-08-18T12:38:32,931][INFO ][o.o.s.c.ConfigurationRepository] [securityBwcCluster0-0] Wait for cluster to be available ...
[2023-08-18T12:38:33,921][WARN ][o.o.d.FileBasedSeedHostsProvider] [securityBwcCluster0-0] expected, but did not find, a dynamic hosts list at [/Users/steecraw/security/bwc-test/build/testclusters/securityBwcCluster0-0/config/unicast_hosts.txt]
[2023-08-18T12:38:33,932][INFO ][o.o.s.c.ConfigurationRepository] [securityBwcCluster0-0] Wait for cluster to be available ...
```
Node then loads configuration and looks for dynamic host lists. It repeats that it is waiting for the cluster be available several times.

**Comment by scrawfor99** [comment]
Then the node starts joining others to form a cluster:

```
[securityBwcCluster0-0] scheduling scheduleNextElection{gracePeriod=0s, thisAttempt=0, maxDelayMillis=100, delayMillis=82, ElectionScheduler{attempt=1, ElectionSchedulerFactory{initialTimeout=100ms, backoffTime=100ms, maxTimeout=10s}}}
[2023-08-18T12:38:37,264][ERROR][o.o.s.a.BackendRegistry  ] [securityBwcCluster0-0] Not yet initialized (you may need to run securityadmin)
[2023-08-18T12:38:37,290][DEBUG][o.o.c.c.ElectionSchedulerFactory] [securityBwcCluster0-0] scheduleNextElection{gracePeriod=0s, thisAttempt=0, maxDelayMillis=100, delayMillis=82, ElectionScheduler{attempt=1, ElectionSchedulerFactory{initialTimeout=100ms, backoffTime=100ms, maxTimeout=10s}}} starting election
[2023-08-18T12:38:37,290][DEBUG][o.o.c.c.ElectionSchedulerFactory] [securityBwcCluster0-0] scheduling scheduleNextElection{gracePeriod=500ms, thisAttempt=1, maxDelayMillis=200, delayMillis=671, ElectionScheduler{attempt=2, ElectionSchedulerFactory{initialTimeout=100ms, backoffTime=100ms, maxTimeout=10s}}}
[2023-08-18T12:38:37,293][DEBUG][o.o.c.c.PreVoteCollector ] [securityBwcCluster0-0] PreVotingRound{preVotesReceived={}, electionStarted=false, preVoteRequest=PreVoteRequest{sourceNode={securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}, currentTerm=0}, isClosed=false} requesting pre-votes from [{securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}, {securityBwcCluster0-1}{frpm-K-xSwmPuqdYsz8U0A}{c9ImaN6YQXOTHENLOPhj7A}{127.0.0.1}{127.0.0.1:54472}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}, {securityBwcCluster0-2}{ID4lsFuCRu6IfHjxbGddTg}{zkPSLl4SSSCNnOl7qKHllw}{127.0.0.1}{127.0.0.1:54477}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}]
[2023-08-18T12:38:37,310][DEBUG][o.o.c.c.PreVoteCollector ] [securityBwcCluster0-0] PreVotingRound{preVotesReceived={{securityBwcCluster0-2}{ID4lsFuCRu6IfHjxbGddTg}{zkPSLl4SSSCNnOl7qKHllw}{127.0.0.1}{127.0.0.1:54477}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}=PreVoteResponse{currentTerm=0, lastAcceptedTerm=0, lastAcceptedVersion=0}, {securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}=PreVoteResponse{currentTerm=0, lastAcceptedTerm=0, lastAcceptedVersion=0}}, electionStarted=false, preVoteRequest=PreVoteRequest{sourceNode={securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}, currentTerm=0}, isClosed=false} added PreVoteResponse{currentTerm=0, lastAcceptedTerm=0, lastAcceptedVersion=0} from {securityBwcCluster0-2}{ID4lsFuCRu6IfHjxbGddTg}{zkPSLl4SSSCNnOl7qKHllw}{127.0.0.1}{127.0.0.1:54477}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}, no quorum yet
[2023-08-18T12:38:37,310][DEBUG][o.o.c.c.PreVoteCollector ] [securityBwcCluster0-0] PreVotingRound{preVotesReceived={{securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}=PreVoteResponse{currentTerm=0, lastAcceptedTerm=0, lastAcceptedVersion=0}}, electionStarted=false, preVoteRequest=PreVoteRequest{sourceNode={securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}, currentTerm=0}, isClosed=false} added PreVoteResponse{currentTerm=0, lastAcceptedTerm=0, lastAcceptedVersion=0} from {securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}, no quorum yet
[2023-08-18T12:38:37,335][DEBUG][o.o.c.c.PreVoteCollector ] [securityBwcCluster0-0] PreVotingRound{preVotesReceived={{securityBwcCluster0-2}{ID4lsFuCRu6IfHjxbGddTg}{zkPSLl4SSSCNnOl7qKHllw}{127.0.0.1}{127.0.0.1:54477}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}=PreVoteResponse{currentTerm=0, lastAcceptedTerm=0, lastAcceptedVersion=0}, {securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}=PreVoteResponse{currentTerm=0, lastAcceptedTerm=0, lastAcceptedVersion=0}, {securityBwcCluster0-1}{frpm-K-xSwmPuqdYsz8U0A}{c9ImaN6YQXOTHENLOPhj7A}{127.0.0.1}{127.0.0.1:54472}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}=PreVoteResponse{currentTerm=0, lastAcceptedTerm=0, lastAcceptedVersion=0}}, electionStarted=true, preVoteRequest=PreVoteRequest{sourceNode={securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}, currentTerm=0}, isClosed=false} added PreVoteResponse{currentTerm=0, lastAcceptedTerm=0, lastAcceptedVersion=0} from {securityBwcCluster0-1}{frpm-K-xSwmPuqdYsz8U0A}{c9ImaN6YQXOTHENLOPhj7A}{127.0.0.1}{127.0.0.1:54472}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}, starting election
[2023-08-18T12:38:37,335][DEBUG][o.o.c.c.Coordinator      ] [securityBwcCluster0-0] starting election with StartJoinRequest{term=1,node={securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}}
[2023-08-18T12:38:37,336][DEBUG][o.o.c.c.Coordinator      ] [securityBwcCluster0-0] joinLeaderInTerm: for [{securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}] with term 1
[2023-08-18T12:38:37,336][DEBUG][o.o.c.c.CoordinationState] [securityBwcCluster0-0] handleStartJoin: leaving term [0] due to StartJoinRequest{term=1,node={securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}}
[2023-08-18T12:38:37,369][DEBUG][o.o.c.c.JoinHelper       ] [securityBwcCluster0-0] attempting to join {securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true} with JoinRequest{sourceNode={securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}, minimumTerm=0, optionalJoin=Optional[Join{term=1, lastAcceptedTerm=0, lastAcceptedVersion=0, sourceNode={securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}, targetNode={securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}}]}
[2023-08-18T12:38:37,373][ERROR][o.o.s.a.BackendRegistry  ] [securityBwcCluster0-0] Not yet initialized (you may need to run securityadmin)
[2023-08-18T12:38:37,371][DEBUG][o.o.c.c.CoordinationState] [securityBwcCluster0-0] handleJoin: added join Join{term=1, lastAcceptedTerm=0, lastAcceptedVersion=0, sourceNode={securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}, targetNode={securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}} from [{securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}] for election, electionWon=false lastAcceptedTerm=0 lastAcceptedVersion=0
[2023-08-18T12:38:37,370][DEBUG][o.o.c.c.JoinHelper       ] [securityBwcCluster0-0] successful response to StartJoinRequest{term=1,node={securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}} from {securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}
[2023-08-18T12:38:37,386][ERROR][o.o.s.a.BackendRegistry  ] [securityBwcCluster0-0] Not yet initialized (you may need to run securityadmin)
[2023-08-18T12:38:37,388][DEBUG][o.o.c.c.JoinHelper       ] [securityBwcCluster0-0] successful response to StartJoinRequest{term=1,node={securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}} from {securityBwcCluster0-1}{frpm-K-xSwmPuqdYsz8U0A}{c9ImaN6YQXOTHENLOPhj7A}{127.0.0.1}{127.0.0.1:54472}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}
[2023-08-18T12:38:37,405][DEBUG][o.o.c.c.CoordinationState] [securityBwcCluster0-0] handleJoin: added join Join{term=1, lastAcceptedTerm=0, lastAcceptedVersion=0, sourceNode={securityBwcCluster0-1}{frpm-K-xSwmPuqdYsz8U0A}{c9ImaN6YQXOTHENLOPhj7A}{127.0.0.1}{127.0.0.1:54472}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}, targetNode={securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}} from [{securityBwcCluster0-1}{frpm-K-xSwmPuqdYsz8U0A}{c9ImaN6YQXOTHENLOPhj7A}{127.0.0.1}{127.0.0.1:54472}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}] for election, electionWon=true lastAcceptedTerm=0 lastAcceptedVersion=0
[2023-08-18T12:38:37,407][DEBUG][o.o.c.c.CoordinationState] [securityBwcCluster0-0] handleJoin: election won in term [1] with VoteCollection{votes=[mWKwveN4RhmQ60MfGRLb6w, frpm-K-xSwmPuqdYsz8U0A], joins=[Join{term=1, lastAcceptedTerm=0, lastAcceptedVersion=0, sourceNode={securityBwcCluster0-1}{frpm-K-xSwmPuqdYsz8U0A}{c9ImaN6YQXOTHENLOPhj7A}{127.0.0.1}{127.0.0.1:54472}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}, targetNode={securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}}, Join{term=1, lastAcceptedTerm=0, lastAcceptedVersion=0, sourceNode={securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}, targetNode={securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}}]}
[2023-08-18T12:38:37,411][DEBUG][o.o.c.c.Coordinator      ] [securityBwcCluster0-0] handleJoinRequest: coordinator becoming LEADER in term 1 (was CANDIDATE, lastKnownLeader was [Optional.empty])
[2023-08-18T12:38:37,422][DEBUG][o.o.c.c.JoinHelper       ] [securityBwcCluster0-0] received a join request for an existing node [{securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}]
[2023-08-18T12:38:37,429][INFO ][o.o.c.s.MasterService    ] [securityBwcCluster0-0] elected-as-cluster-manager ([2] nodes joined)[{securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true} elect leader, {securityBwcCluster0-1}{frpm-K-xSwmPuqdYsz8U0A}{c9ImaN6YQXOTHENLOPhj7A}{127.0.0.1}{127.0.0.1:54472}{dimr}{testattr=test, shard_indexing_pressure_enabled=true} elect leader, _BECOME_CLUSTER_MANAGER_TASK_, _FINISH_ELECTION_], term: 1, version: 1, delta: cluster-manager node changed {previous [], current [{securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}]}, added {{securityBwcCluster0-1}{frpm-K-xSwmPuqdYsz8U0A}{c9ImaN6YQXOTHENLOPhj7A}{127.0.0.1}{127.0.0.1:54472}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}}
[2023-08-18T12:38:37,439][DEBUG][o.o.c.c.PublicationTransportHandler] [securityBwcCluster0-0] received full cluster state version [1] with size [429]
[2023-08-18T12:38:37,457][DEBUG][o.o.c.c.JoinHelper       ] [securityBwcCluster0-0] successful response to StartJoinRequest{term=1,node={securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}} from {securityBwcCluster0-2}{ID4lsFuCRu6IfHjxbGddTg}{zkPSLl4SSSCNnOl7qKHllw}{127.0.0.1}{127.0.0.1:54477}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}
[2023-08-18T12:38:37,498][ERROR][o.o.s.a.BackendRegistry  ] [securityBwcCluster0-0] Not yet initialized (you may need to run securityadmin)
[2023-08-18T12:38:37,498][INFO ][o.o.c.c.CoordinationState] [securityBwcCluster0-0] cluster UUID set to [Vw4CIyDYRpaLD1d9WnpxQw]
```

For some reason 0-2 is listed twice while 0-1 is not listed. 

The node then states that it successfully joined itself:

```
[securityBwcCluster0-0] cluster-manager node changed {previous [], current [{securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}]}, added {{securityBwcCluster0-1}{frpm-K-xSwmPuqdYsz8U0A}{c9ImaN6YQXOTHENLOPhj7A}{127.0.0.1}{127.0.0.1:54472}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}}, term: 1, version: 1, reason: Publication{term=1, version=1}
[2023-08-18T12:38:37,565][DEBUG][o.o.c.c.C.CoordinatorPublication] [securityBwcCluster0-0] publication ended successfully: Publication{term=1, version=1}
[2023-08-18T12:38:37,569][DEBUG][o.o.c.c.JoinHelper       ] [securityBwcCluster0-0] successfully joined {securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true} with JoinRequest{sourceNode={securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}, minimumTerm=0, optionalJoin=Optional[Join{term=1, lastAcceptedTerm=0, lastAcceptedVersion=0, sourceNode={securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}, targetNode={securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}}]}
[2023-08-18T12:38:37,569][INFO ][o.o.d.PeerFinder         ] [securityBwcCluster0-0] setting findPeersInterval to [1s] as node commission status = [true] for local node [{securityBwcCluster0-0}{mWKwveN4RhmQ60MfGRLb6w}{BKgovVncSY2hgwndsgq7sQ}{127.0.0.1}{127.0.0.1:54466}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}]
[2023-08-18T12:38:37,570][INFO ][o.o.c.s.MasterService    ] [securityBwcCluster0-0] node-join[{securityBwcCluster0-2}{ID4lsFuCRu6IfHjxbGddTg}{zkPSLl4SSSCNnOl7qKHllw}{127.0.0.1}{127.0.0.1:54477}{dimr}{testattr=test, shard_indexing_pressure_enabled=true} join existing leader], term: 1, version: 2, delta: added {{securityBwcCluster0-2}{ID4lsFuCRu6IfHjxbGddTg}{zkPSLl4SSSCNnOl7qKHllw}{127.0.0.1}{127.0.0.1:54477}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}}
[2023-08-18T12:38:37,571][DEBUG][o.o.c.c.PublicationTransportHandler] [securityBwcCluster0-0] received full cluster state version [2] with size [478]
[2023-08-18T12:38:37,573][INFO ][o.o.c.r.a.DiskThresholdMonitor] [securityBwcCluster0-0] skipping monitor as a check is already in progress
[2023-08-18T12:38:37,607][ERROR][o.o.s.a.BackendRegistry  ] [securityBwcCluster0-0] Not yet initialized (you may need to run securityadmin)
[2023-08-18T12:38:37,646][INFO ][o.o.c.s.ClusterApplierService] [securityBwcCluster0-0] added {{securityBwcCluster0-2}{ID4lsFuCRu6IfHjxbGddTg}{zkPSLl4SSSCNnOl7qKHllw}{127.0.0.1}{127.0.0.1:54477}{dimr}{testattr=test, shard_indexing_pressure_enabled=true}}, term: 1, version: 2, reason: Publication{term=1, version=2}
[2023-08-18T12:38:37,647][DEBUG][o.o.c.c.C.CoordinatorPublication] [securityBwcCluster0-0] publication ended successfully: Publication{term=1, version=2}
[2023-08-18T12:38:37,648][DEBUG][o.o.c.c.PublicationTransportHandler] [securityBwcCluster0-0] received full cluster state version [3] with size [480]
[2023-08-18T12:38:37,718][ERROR][o.o.s.a.BackendRegistry  ] [securityBwcCluster0-0] Not yet initialized (you may need to run securityadmin)
[2023-08-18T12:38:37,768][DEBUG][o.o.c.c.C.CoordinatorPublication] [securityBwcCluster0-0] publication ended successfully: Publication{term=1, version=3}
[2023-08-18T12:38:37,774][DEBUG][o.o.c.c.PublicationTransportHandler] [securityBwcCluster0-0] received full cluster state version [4] with size [403]
[2023-08-18T12:38:37,808][DEBUG][o.o.c.c.C.CoordinatorPublication] [securityBwcCluster0-0] publication ended successfully: Publication{term=1, version=4}
[2023-08-18T12:38:37,808][INFO ][o.o.g.GatewayService     ] [securityBwcCluster0-0] recovered [0] indices into cluster_state
```

BackendRegistry continually states that cluster is not initialized while 
```
Setting replication.type: DOCUMENT will be used for Index until Segment Replication supports System and Hidden indices
[2023-08-18T12:38:37,962][INFO ][o.o.p.PluginsService     ] [securityBwcCluster0-0] PluginService:onIndexModule index:[.opendistro_security/OdEfyqiYRI-9sXGvFapsFQ]
[2023-08-18T12:38:37,966][DEBUG][o.o.c.c.ElectionSchedulerFactory] [securityBwcCluster0-0] scheduleNextElection{gracePeriod=500ms, thisAttempt=1, maxDelayMillis=200, delayMillis=671, ElectionScheduler{attempt=2, ElectionSchedulerFactory{initialTimeout=100ms, backoffTime=100ms, maxTimeout=10s}}} not starting election
[2023-08-18T12:38:38,009][INFO ][o.o.c.m.MetadataCreateIndexService] [securityBwcCluster0-0] [.opendistro_security] creating index, cause [api], templates [], shards [1]/[1]
[2023-08-18T12:38:38,015][INFO ][o.o.c.r.a.AllocationService] [securityBwcCluster0-0] updating number_of_replicas to [2] for indices [.opendistro_security]
[2023-08-18T12:38:38,027][DEBUG][o.o.c.c.PublicationTransportHandler] [securityBwcCluster0-0] received diff cluster state version [5] with uuid [8c_0gAWoSS2fVbHVwsofgg], diff size [460]
[2023-08-18T12:38:38,053][ERROR][o.o.s.a.BackendRegistry  ] [securityBwcCluster0-0] Not yet initialized (you may need to run securityadmin)
[2023-08-18T12:38:38,163][ERROR][o.o.s.a.BackendRegistry  ] [securityBwcCluster0-0] Not yet initialized (you may need to run securityadmin)
[2023-08-18T12:38:38,202][DEBUG][o.o.c.c.C.CoordinatorPublication] [securityBwcCluster0-0] publication ended successfully: Publication{term=1, version=5}
[2023-08-18T12:38:38,267][ERROR][o.o.s.a.BackendRegistry  ] [securityBwcCluster0-0] Not yet initialized (you may need to run securityadmin)
[2023-08-18T12:38:38,329][DEBUG][o.o.c.c.PublicationTransportHandler] [securityBwcCluster0-0] received diff cluster state version [6] with uuid [OWHQUrogQoino33iAS3ayA], diff size [462]
[2023-08-18T12:38:38,372][ERROR][o.o.s.a.BackendRegistry  ] [securityBwcCluster0-0] Not yet initialized (you may need to run securityadmin)
[2023-08-18T12:38:38,394][INFO ][o.o.s.c.ConfigurationRepository] [securityBwcCluster0-0] Index .opendistro_security created?: true
[2023-08-18T12:38:38,394][DEBUG][o.o.c.c.C.CoordinatorPublication] [securityBwcCluster0-0] publication ended successfully: Publication{term=1, version=6}
[2023-08-18T12:38:38,397][INFO ][o.o.s.c.ConfigurationRepository] [securityBwcCluster0-0] Node started, try to initialize it. Wait for at least yellow cluster state....
[2023-08-18T12:38:38,402][DEBUG][o.o.c.c.PublicationTransportHandler] [securityBwcCluster0-0] received diff cluster state version [7] with uuid [gkvmh1GoQly6eZpSTAfEiQ], diff size [321]
[2023-08-18T12:38:38,403][INFO ][o.o.s.s.ConfigHelper     ] [securityBwcCluster0-0] Will update 'config' with /Users/steecraw/security/bwc-test/build/testclusters/securityBwcCluster0-0/config/opensearch-security/config.yml and populate it with empty doc if file missing and populateEmptyIfFileMissing=false
[2023-08-18T12:38:38,433][INFO ][o.o.p.PluginsService     ] [securityBwcCluster0-0] PluginService:onIndexModule index:[.opendistro_security/OdEfyqiYRI-9sXGvFapsFQ]
[2023-08-18T12:38:38,470][DEBUG][o.o.c.c.C.CoordinatorPublication] [securityBwcCluster0-0] publication ended successfully: Publication{term=1, version=7}
[2023-08-18T12:38:38,475][DEBUG][o.o.c.c.PublicationTransportHandler] [securityBwcCluster0-0] received diff cluster state version [8] with uuid [Hx22rt6pR26uKZscyRNQ3Q], diff size [341]
[2023-08-18T12:38:38,491][ERROR][o.o.s.a.BackendRegistry  ] [securityBwcCluster0-0] Not yet initialized (you may need to run securityadmin)
[2023-08-18T12:38:38,502][ERROR][o.o.s.a.BackendRegistry  ] [securityBwcCluster0-0] Not yet initialized (you may need to run securityadmin)
[2023-08-18T12:38:38,613][ERROR][o.o.s.a.BackendRegistry  ] [securityBwcCluster0-0] Not yet initialized (you may need to run securityadmin)
[2023-08-18T12:38:38,624][DEBUG][o.o.c.c.C.CoordinatorPublication] [securityBwcCluster0-0] publication ended successfully: Publication{term=1, version=8}
[2023-08-18T12:38:38,629][INFO ][o.o.p.PluginsService     ] [securityBwcCluster0-0] PluginService:onIndexModule index:[.opendistro_security/OdEfyqiYRI-9sXGvFapsFQ]
[2023-08-18T12:38:38,640][INFO ][o.o.c.m.MetadataMappingService] [securityBwcCluster0-0] [.opendistro_security/OdEfyqiYRI-9sXGvFapsFQ] create_mapping
[2023-08-18T12:38:38,645][DEBUG][o.o.c.c.PublicationTransportHandler] [securityBwcCluster0-0] received diff cluster state version [9] with uuid [vecMUuvrQ9SE97jvsrpWbA], diff size [532]
[2023-08-18T12:38:38,724][ERROR][o.o.s.a.BackendRegistry  ] [securityBwcCluster0-0] Not yet initialized (you may need to run securityadmin)
[2023-08-18T12:38:38,742][DEBUG][o.o.c.c.C.CoordinatorPublication] [securityBwcCluster0-0] publication ended successfully: Publication{term=1, version=9}
```

**Comment by scrawfor99** [comment]
I opened a manual backport against 2.x in core.

**Comment by scrawfor99** [comment]
I was able to get all tests passing on a change from 2.8 to 2.9. 

Here are the steps to reproduce:

Use this build of 2.9: https://github.com/scrawfor99/OpenSearch/tree/2.9.0
and this build of 2.8: https://github.com/scrawfor99/OpenSearch/tree/2.8.0

For each of these, I cherry picked the commits to my backport in the 2.x line which can be found here: https://github.com/opensearch-project/OpenSearch/pull/9444. 

I then ran `./gradlew localDistro` and `./gradlew publishToMavenLocal` for each. 
I then went to the security branch here: https://github.com/scrawfor99/security/tree/working2BWC and followed the steps used previously but replacing the earlier version for the `build.gradle` BWC test file with 2.8 and the later with 2.9. 

Finally I ran:

```
./gradlew -p bwc-test bwcTestSuite -Dtests.opensearch.secure=true -Dtests.opensearch.username=admin -Dtests.opensearch.password=admin -PcustomDistributionUrl="${GIT_PROJECT_ROOT}/OpenSearch/distribution/archives/linux-tar/build/distributions/opensearch-min-2.8.0-SNAPSHOT-linux-x64.tar.gz" -i

```
Which grabbed the localDistro of the earlier version file which I had created. I am skeptical that part is truly necessary however.

Here are the test results parsed from the output: 

```
> Task :securityBwcCluster#mixedClusterTaskrTask0
Caching disabled for task ':securityBwcCluster#mixedClusterTask' because:ecurity/bwc-test/build/test-results/securityBwcCluster#oldVersionClusterTask0
  Build cache is disablededClusterTaskrTask0 > 1 test completedrg.gradle.api.internal.tasks.testing.worker.TestWorkerityIT

> Task :securityBwcCluster#fullRestartClusterTask
Caching disabled for task ':securityBwcCluster#fullRestartClusterTask' because:y/bwc-test/build/test-results/securityBwcCluster#oldVersionClusterTask1
  Build cache is disabledlRestartClusterTask > 1 test completedrg.gradle.api.internal.tasks.testing.worker.TestWorkerityIT

Task :securityBwcCluster#rollingUpgradeClusterTask
Finished generating test XML results (0.0 secs) into: /Users/steecraw/security/bwc-test/build/test-results/securityBwcCluster#rollingUpgradeClusterTask
Generating HTML test report...pgradeClusterTask > 1 test completedgradle.api.internal.tasks.testing.worker.TestWorkerbilityITe/index].

```
This appears to me that all tests pass since the assertions of the tests are all based around calling the SecurityBackwardsCompatabilityIT:

```
org.opensearch.security.bwc.SecurityBackwardsCompatibilityIT STANDARD_OUTrch.security.bwc.SecurityBackwardsCompatibilityIT
    [2023-08-18T12:38:22,482][WARN ][o.o.b.Natives            ] [[SUITE-SecurityBackwardsCompatibilityIT-seed#[1934D36967F83B97]]] unable to load JNA native support library, na
tive methods will be disabled./work -Dorg.gradle.native=false -Dtests.artifact=bwc-test -Dtests.clustername=securityBwcCluster1 -Dtests.gradle=
```

**Comment by scrawfor99** [comment]
Blocked pending direction forward -- do we want to work on fixing whatever is wrong with main or prioritize merging (https://github.com/opensearch-project/security/pull/2802)?

**Comment by scrawfor99** [comment]
Going to be splitting this into a 2.x fix and a main fix for now. That will mean that the changes in the working 2.x will be submitted as a PR to the 2.x line and then we will follow up on main from a separate issue.
